src/store/selectors/deck-filters.ts

```diff
@@ -0,0 +1,277 @@
+  state.deckFilters.filters.search;
+
+export const selectSearchableTextInDecks = createSelector(
+  selectLocalDecks,
+  (decks) => {
+    // deck.investigator_name doesn't work for imported decks?
+    return decks.map(
+      (deck) =>
+        `${deck.name}|${deck.investigator_name || deck.cards.investigator.card.real_name}`,
+    );
+  },
+);
+
+// Faction
+export const selectDeckFactionFilter = (state: StoreState) =>
+  state.deckFilters.filters.faction;
+
+const filterDeckByFaction = (faction: string) => {
+  return (deck: ResolvedDeck) => {
+    return deck.cards.investigator.card.faction_code === faction;
+  };
+};
+
+const makeDeckFactionFilter = (values: MultiselectFilter) => {
+  return or(values.map((value) => filterDeckByFaction(value)));
+};
+
+// Tag
+const filterDeckByTag = (tag: string) => {
+  return (deck: ResolvedDeck) => {
+    return deck.tags.includes(tag);
+  };
+};
+
+const makeDeckTagsFilter = (values: MultiselectFilter) => {
+  return or(values.map((value) => filterDeckByTag(value)));
+};
+
+export const selectTagsChanges = createSelector(
+  selectDeckFilters,
+  (filters) => {
+    const tagsFilters = filters.tags;
+    if (!tagsFilters.length) return "";
+    return tagsFilters.map(capitalize).join(" or ");
+  },
+);
+
+// Properties
+export const selectDeckPropertiesFilter = (state: StoreState) =>
+  state.deckFilters.filters.properties;
+
+export const selectPropertiesChanges = createSelector(
+  selectDeckPropertiesFilter,
+  (properties) => {
+    return Object.keys(properties)
+      .filter((prop) => properties[prop as DeckPropertyName])
+      .map(capitalize)
+      .join(" and ");
+  },
+);
+
+const makeDeckPropertiesFilter = (
+  properties: Record<DeckPropertyName, boolean>,
+) => {
+  const filters = [];
+  for (const property of Object.keys(properties)) {
+    if (properties[property as DeckPropertyName]) {
+      switch (property) {
+        case "parallel":
+          filters.push((deck: ResolvedDeck) =>
+            Boolean(
+              deck.investigatorFront.card.parallel ||
+                deck.investigatorBack.card.parallel,
+            ),
+          );
+      }
+    }
+  }
+  return and(filters);
+};
+
+// Validity
+const makeDeckValidityFilter = (value: Omit<DeckValidity, "all">) => {
+  switch (value) {
+    case "valid":
+      return (deck: ResolvedDeck) => deck.problem === null;
+    case "invalid":
+      return (deck: ResolvedDeck) => Boolean(deck.problem);
+    default:
+      return (_: ResolvedDeck) => true;
+  }
+};
+
+// Exp Cost
+export const selectDecksMinMaxExpCost = createSelector(
+  selectLocalDecks,
+  (decks) => {
+    const minmax: RangeMinMax = decks.reduce<[number, number]>(
+      (acc, val) => {
+        const { xpRequired } = val.stats;
+        acc[0] = Math.min(acc[0], xpRequired);
+        acc[1] = Math.max(acc[1], xpRequired);
+        return acc;
+      },
+      [Number.POSITIVE_INFINITY, Number.NEGATIVE_INFINITY],
```

Comment:
Are these switched around or am I misreading the intent? xp can't be negative, so I would vote for a 0 baseline

Reply:
The intent is to get the min and max xp required of decks in your collection, right? so, imagine if you have _only_ 3 decks with xp costs of 10, 20, and 30 - the range should be 10-30, so for the first value, the minimum should be the maximum possible number (like, what if you only have 5 invalid decks of 999xp? 🤔). For the maximum, I agree, the baseline can be 0.